common.sh

```diff
@@ -187,9 +187,29 @@ create_instance()
 get_instance_ip()
 {
     execution_seq=$((${execution_seq}+1))
-    INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
-                        --query "Reservations[*].Instances[*].PrivateIpAddress" \
-                        --output=text)
+    local retry=20
+    local ret=0
+    set +e
+    while [ $retry -ge 0 ]; do
+        INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
+                            --query "Reservations[*].Instances[*].PrivateIpAddress" \
+                            --output=text)
+        ret=$?
+        if [ "$ret" -eq 0 ] && [ -n "$INSTANCE_IPS" ]; then
+            echo "get instance ips successfully."
+            break
+        else
+            echo "describe-instances fails or does not return instance ips, retry after 10 seconds."
+            sleep 10
+        fi
+        retry=$((retry-1))
+    done
+    if [ "$retry" -eq 0 ]; then
```

Comment:
This seems like the wrong pattern.  Check to see if we have instance IPs; that's the real condition we care about.

Reply:
Thanks, fixed in the latest revision.